dts/bindings/adc/adc-controller.yaml

```diff
@@ -12,3 +13,102 @@ properties:
     "#io-channel-cells":
       type: int
       required: true
+
+    "#address-cells":
+      required: false
+      const: 1
+
+    "#size-cells":
+      required: false
+      const: 0
+
+child-binding:
+    description: |
+      Channel configuration. See the adc_channel_cfg structure in adc.h.
+
+      All nodes using this binding must be named "channel", otherwise their
+      data will not be accessible by the ADC_CHANNEL_DT_NODE macro.
+
+    properties:
+      reg:
+        type: array
+        required: true
+        description: Channel identifier.
+
+      gain:
+        type: string
+        required: true
+        description: Gain selection.
+        enum:
+          - "ADC_GAIN_1_6" # x 1/6.
```

Comment:
Remember that comments get lost unless you read the binding file itself. If you put this information in the binding description instead, it will be available in the devicetree bindings index in the documentation.